Examples/Physics_applications/laser_acceleration/inputs_2d_rz

```diff
@@ -85,3 +85,4 @@ laser1.wavelength = 0.8e-6               # The wavelength of the laser (in m)
 diagnostics.diags_names = diag1
 diag1.period = 200
 diag1.diag_type = Full
+diag1.fields_to_plot = Ex Ey Ez Bx By Bz jx jy jz rho
```

Comment:
looks to me like diagnostics for RZ and rho does crash at runtime right now:
```
 3: ./main2d.gnu.TEST.TPROF.MPI.OMP.RZ.ex(+0x197e1f) [0x557280f01e1f]
    doChargeDepositionShapeN<3>(GetParticlePosition const&, double const*, int const*, amrex::FArrayBox&, long, std::array<double, 3ul> const&, std::array<double, 3ul>, amrex::Dim3, double, long)::{lambda(long)#1}::operator()(long) const
/tmp/ci-zQmUG9Gm5u/amrex//Src/Base/AMReX_GpuAtomic.H:123:14

 4: ./main2d.gnu.TEST.TPROF.MPI.OMP.RZ.ex(+0x18bc5f) [0x557280ef5c5f]
    void amrex::ParallelFor<long, doChargeDepositionShapeN<3>(GetParticlePosition const&, double const*, int const*, amrex::FArrayBox&, long, std::array<double, 3ul> const&, std::array<double, 3ul>, amrex::Dim3, double, long)::{lambda(long)#1}, void>(long, doChargeDepositionShapeN<3>(GetParticlePosition const&, double const*, int const*, amrex::FArrayBox&, long, std::array<double, 3ul> const&, std::array<double, 3ul>, amrex::Dim3, double, long)::{lambda(long)#1}&&, unsigned long) inlined at ./Source/Particles/Deposition/ChargeDeposition.H:66:23 in WarpXParticleContainer::DepositCharge(WarpXParIter&, amrex::PODVector<double, std::allocator<double> >&, int const*, amrex::MultiFab*, int, long, long, int, int, int)
/tmp/ci-zQmUG9Gm5u/amrex//Src/Base/AMReX_GpuLaunchFunctsC.H:24:17
void doChargeDepositionShapeN<3>(GetParticlePosition const&, double const*, int const*, amrex::FArrayBox&, long, std::array<double, 3ul> const&, std::array<double, 3ul>, amrex::Dim3, double, long)
./Source/Particles/Deposition/ChargeDeposition.H:66:23
WarpXParticleContainer::DepositCharge(WarpXParIter&, amrex::PODVector<double, std::allocator<double> >&, int const*, amrex::MultiFab*, int, long, long, int, int, int)
./Source/Particles/WarpXParticleContainer.cpp:480:36

 5: ./main2d.gnu.TEST.TPROF.MPI.OMP.RZ.ex(+0x18c560) [0x557280ef6560]
    WarpXParticleContainer::GetChargeDensity(int, bool) [clone ._omp_fn.20]
/tmp/ci-zQmUG9Gm5u/amrex//Src/Particle/AMReX_ParIter.H:48:9

 6: /usr/lib/x86_64-linux-gnu/libgomp.so.1(GOMP_parallel+0x3f) [0x7f67ec549edf]
    GOMP_parallel
??:0
```

maybe cc @dpgrote @EZoni: have you seen something similar?

Reply:
The problem comes from the fact that we go out of bound on the fourth component of the MultiFab storing `rho`, inside the following loop:
https://github.com/ECP-WarpX/WarpX/blob/345feb7faa0647ec52025adb450c2855154e8111/Source/Particles/Deposition/ChargeDeposition.H#L138-L142
In particular, we have `imode=1` and try to access the fourth components with index `2*imode-1=1` and `2*imode=2`, while the array `rho_arr` has the following bounds: `(-3:67,493:507,0:0,0:0)`.

The reason why the array `rho_arr` has only 1 element in the fourth component, with index `0`, is that it comes from the FArrayBox `rho_fab` allocated here:
https://github.com/ECP-WarpX/WarpX/blob/345feb7faa0647ec52025adb450c2855154e8111/Source/Particles/WarpXParticleContainer.cpp#L438-L443
In turn, the number of components `nc` along the fourth direction, used to resize `local_rho`, is set here:
https://github.com/ECP-WarpX/WarpX/blob/345feb7faa0647ec52025adb450c2855154e8111/Source/Particles/WarpXParticleContainer.cpp#L429
Here, the number of components of `rho` is correctly `rho->nComp()=3` for `n_rz_azimuthal_modes=2`, but after the division by `2` above we get `nc=1`.

It seems to me that we would like to have at least `nc=rho->nComp()=n_rz_azimuthal_modes*2-1` components, instead of `nc=rho->nComp()/2`, along the fourth direction (so that we can access all components up to index `2*imode`), in order to deposit all azimuthal modes correctly. @dpgrote Do you agree with this? If yes, I can open a small PR to fix this.

Reply:
fix: #1150 :+1: